cloudwatch_logs_common/src/utils/log_file_manager.cpp

```diff
@@ -54,6 +65,11 @@ FileObject<LogCollection> LogFileManager::readBatch(
     log_set.insert(input_event);
     data_tokens.push_back(data_token);
   }
+
+  if(!validateLogTime(*log_set.begin(), *log_set.end())){
```

Comment:
Something else to consider: how to we separate the data so we can attempt to send later? Otherwise when reading the same data block we may keep running into the issue. I don't remember the details of the `DataToken` but we'll want to make sure the behavior is ok (either read more data to meet the batch size or exit the loop prematurely and try to send less data than desired).

You might want to make this check while reading the data in the for loop above.